WebApiCRUD/Controllers/TodoItemController.cs

```diff
@@ -0,0 +1,220 @@
+                await _context.SaveChangesAsync();
+
+  
+                var response = new
+                {
+                    Message = $"Todo item with ID {todo.Id} created successfully"
+                };
+
+                return Ok(response); // Return status 200 
+            }
+            else
+            {
+                return BadRequest(ModelState); // Return status 400
+            }
+        }
+
+        // PUT /api/todo/{id}
+        [HttpPut("{id}")]
+        [Authorize(Roles = "User, Admin")] 
+        public async Task<IActionResult> UpdateTodo(Guid id, TodoItem updatedTodo)
+        {
+
+            int? userId = null;
+            var userIdClaim = _httpContextAccessor.HttpContext.User.FindFirst("id")?.Value;
+            if (userIdClaim != null && int.TryParse(userIdClaim, out int parsedUserId))
+            {
+                userId = parsedUserId;
+            }
+
+            var existingTodo = await _context.Todos.FindAsync(id);
+
+            if (existingTodo == null)
+            {
+                return NotFound(); // Return status 404 if the todo item doesn't exist
+            }
+
+            // Get the user's roles from the claims
+            var userRoles = _httpContextAccessor.HttpContext.User.FindAll(ClaimTypes.Role).Select(c => c.Value).ToList();
+
+            if (userRoles.Contains("Admin") || (userRoles.Contains("User") && existingTodo.UserId == userId))
+            {
+                existingTodo.Title = updatedTodo.Title;
+                existingTodo.IsCompleted = updatedTodo.IsCompleted;
+                existingTodo.UpdatedDate = DateTime.UtcNow;
+
+                await _context.SaveChangesAsync();
+
+                var response = new
+                {
+                    Message = $"Todo item with ID {id} updated successfully"
+                };
+
+                return Ok(response);
+            }
+            else
+            {
+                return Forbid(); // User is not authorized to update the record
+            }
+        }
+
+        // DELETE /api/todo/{id}
+        [HttpDelete("{id}")]
+        [Authorize(Roles = "User, Admin")]
+        public async Task<IActionResult> DeleteTodo(Guid id)
+        {
+            int? userId = null;
+            var userIdClaim = _httpContextAccessor.HttpContext.User.FindFirst("id")?.Value;
+            if (userIdClaim != null && int.TryParse(userIdClaim, out int parsedUserId))
+            {
+                userId = parsedUserId;
+            }
+
+            var todo = await _context.Todos.FindAsync(id);
+
+            if (todo == null)
+            {
+                return NotFound(); // Return status 404 if the todo item doesn't exist
+            }
+
+            // Get the user's roles from the claims
+            var userRoles = _httpContextAccessor.HttpContext.User.FindAll(ClaimTypes.Role).Select(c => c.Value).ToList();
+
+            if (userRoles.Contains("Admin") || (userRoles.Contains("User") && todo.UserId == userId))
```

Comment:
You should implement a middleware to implement the logic "One user should only perform CRUD over its own ToDos". If a particular Todo is not under ownership of a user then it Middleware should intercept this call and it should not reach the Controller"